django-stubs/db/models/functions/datetime.pyi

```diff
@@ -33,7 +38,21 @@ class TruncBase(TimezoneMixin, Transform):
     kind: str
     tzinfo: Any
 
-class Trunc(TruncBase): ...
+    def __init__(
+        self, expression: Expression, output_field: Field | None = ..., tzinfo: tzinfo | None = ..., **extra: Any
+    ) -> None: ...
+    def as_sql(self, compiler: SQLCompiler, connection: BaseDatabaseWrapper) -> _AsSqlType: ...  # type: ignore[override]
+
+class Trunc(TruncBase):
+    def __init__(
+        self,
+        expression: Expression,
```

Comment:
the types on `expression` do not match `django.db.expressions.Func` which currently has this typed as `Any` -- notably these expressions go through parsing and (at the very least) accept `str` as well